boards/xiao_m0/src/lib.rs

```diff
@@ -1,160 +1,204 @@
 #![no_std]
 
 pub use atsamd_hal as hal;
-pub use hal::common::*;
-pub use hal::pac;
-
 #[cfg(feature = "rt")]
 pub use cortex_m_rt::entry;
-
-use hal::prelude::*;
-use hal::{
-    clock::GenericClockController,
-    define_pins,
-    gpio::PfD,
-    gpio::{Floating, Input, Port},
-    pad::PadPin,
-    sercom::{I2CMaster2, SPIMaster0, UART4},
-    time::Hertz,
+use hal::clock::GenericClockController;
+pub use hal::ehal;
+pub use hal::pac;
+use hal::sercom::{
+    v2::{uart, Sercom0, Sercom4},
+    I2CMaster0,
 };
-
+use hal::time::Hertz;
 #[cfg(feature = "usb")]
-use hal::gpio::v2::{AnyPin, PA24, PA25};
-#[cfg(feature = "usb")]
-use hal::usb::usb_device::bus::UsbBusAllocator;
-#[cfg(feature = "usb")]
-pub use hal::usb::UsbBus;
-
-define_pins!(
-    struct Pins,
-    pac: pac,
-
-    /// Pin A0/D0/DAC
-    pin a0 = a2,
-    /// Pin A1/D1
-    pin a1 = a4,
-    /// Pin A2/D2
-    pin a2 = a10,
-    /// Pin A3/D3
-    pin a3 = a11,
-    /// Pin A4/D4/SDA
-    pin a4 = a8,
-    /// Pin A5/D5/SCL
-    pin a5 = a9,
-    /// Pin A6/D6/TX
-    pin a6 = b8,
-    /// Pin A7/D7/RX
-    pin a7 = b9,
-    /// Pin A8/D8/SCK
-    pin a8 = a7,
-    /// Pin A9/D9/MISO
-    pin a9 = a5,
-    /// Pin A10/D10/MOSI
```

Comment:
could we keep these doc comments on the pins?

Reply:
I reverted these names - could you explain to e where they are actually used/visible? I don't see these macros expanded in my IDE so they weren't useful for me

Reply:
Having the docstrings allows documentation like this to be produced: https://docs.rs/feather_m4/0.8.0/feather_m4/struct.Pins.html#structfield.a0